File: wheystore/wheystore.Web/Modules/SanPhamDB/QuaTang/QuaTangColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.QuaTang")]
[BasedOnRow(typeof(QuaTangRow), CheckNames = true)]
public class QuaTangColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string TenQua { get; set; }
    public decimal GiaTri { get; set; }
    public string MoTa { get; set; }
    public string Anh { get; set; }
}

File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/QuaTangColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { QuaTangRow } from "./QuaTangRow";

export interface QuaTangColumns {
    Id: Column<QuaTangRow>;
    TenQua: Column<QuaTangRow>;
    GiaTri: Column<QuaTangRow>;
    MoTa: Column<QuaTangRow>;
    Anh: Column<QuaTangRow>;
}

export class QuaTangColumns extends ColumnsBase<QuaTangRow> {
    static readonly columnsKey = 'SanPhamDB.QuaTang';
    static readonly Fields = fieldsProxy<QuaTangColumns>();
}